### PR TITLE
Project improvements

### PR DESCRIPTION
**Pull request by Codex-**
Just a small set of project improvements based on what I saw at first glance when going to implement some features.

- @types are dev deps, not real deps, unless exported explicitly as a part of a package to be consumed.
- Updated the gitignore with things that were immediately generated and then git tracked.
- Abstracted the eslint rules out to a dedicated `.eslintrc.json`
  - Sorted / organised the file.
  - Added section to handle the JS config explicitly.
    - Means you can avoid the copy paste ignores.
  - Added `@typescript-eslint/await-thenable` & `@typescript-eslint/no-floating-promises`.
    - These two warn on what in my experience is one of the most common sources for bugs in applications.
- Improve the CICD workflows.
  - Bump to use latest `setup-node`.
  - Run lint / build and every commit.
  - Only generate artifacts for PR's (original behaviour).
- Update use of deprecated webpack config.

### CODE REVIEW

**Comment by Codex-** [comment]
Rebased on webui v2 branch.

File: src/core/middlewares/signalr.ts [hunk 1]
```diff
@@ -121,8 +121,8 @@ const signalRMiddleware = ({
     connectionLog.on('Log', onLogsLog(dispatch));
     startSignalRConnection(connectionLog);
   } else if (action.type === Events.AUTH_LOGOUT) {
-    if (connectionEvents !== undefined) connectionEvents.stop();
-    if (connectionLog !== undefined) connectionLog.stop();
+    await connectionEvents?.stop();
+    await connectionLog?.stop();
```
Comment (Codex-) [inline src/core/middlewares/signalr.ts:125]: If you want to float these promises then consider using an IIFE to make it explicitly clear that a floated promise is the intention, that is the intention behind how the eslint rule is configured.

**Review by hidden4003 — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
         with:
           node-version: 14
       - name: Install dependencies
-        run: yarn install --frozen-lockfile
+        run: yarn install --immutable-cache # lock changes should already be committed.
```
Comment (hidden4003) [inline .github/workflows/CI.yml:18]: docs say you are supposed to use --immutable as a replacement for --frozen-lockfile, not --immutable-cache

File: .github/workflows/Release.yml [hunk 3]
```diff
@@ -12,11 +12,11 @@ jobs:
...
         uses: actions/checkout@v2
       - name: Setup Node.js v14
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
-          node-version: '14'
+          node-version: 14
       - name: Install dependencies
-        run: yarn install --production=false
+        run: yarn install --frozen-lockfile --production=false
```
Comment (hidden4003) [inline .github/workflows/Release.yml:19]: needs to be --immutable here as well

**Review by hidden4003 — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml (hunk 2 above)
Comment (Codex-) [reply .github/workflows/CI.yml:18]: `immutable` is default enabled in CI environments:
```
If the --immutable option is set (defaults to true on CI)
```

https://yarnpkg.com/cli/install
image

`--immutable-cache` is more desirable behaviour for ensuring that CICD is running the intended and committed state of the repository.

File: .github/workflows/Release.yml (hunk 3 above)
Comment (Codex-) [reply .github/workflows/Release.yml:19]: Ah damn, will update this one. Probably shouldn't have the `production` modifier either

File: .github/workflows/CI.yml (hunk 2 above)
Comment (Codex-) [reply .github/workflows/CI.yml]: Hmm odd, removed for now.

**Pull request merged**